src/cpp/processor/Processor.cpp

```diff
@@ -45,8 +45,24 @@ template<typename EndPoint>
 void Processor<EndPoint>::process_input_packet(
         InputPacket<EndPoint>&& input_packet)
 {
-    if ((input_packet.message->get_header().session_id() == dds::xrce::SESSIONID_NONE_WITH_CLIENT_KEY) ||
-        (input_packet.message->get_header().session_id() == dds::xrce::SESSIONID_NONE_WITHOUT_CLIENT_KEY))
+    dds::xrce::MessageHeader header = input_packet.message->get_header();
+    dds::xrce::ClientKey client_key = dds::xrce::CLIENTKEY_INVALID;
+    if (128 > header.session_id())
+    {
+        client_key = header.client_key();
+    }
+    else
+    {
+        uint32_t raw_client_key;
+        if (server_.get_client_key(input_packet.source, raw_client_key))
+        {
+            client_key = conversion::raw_to_clientkey(raw_client_key);
+        }
+    }
+    std::shared_ptr<ProxyClient> client = root_.get_client(client_key);
```

Comment:
This could stay in the `else` statement

Reply:
789d7df13d123b7253cca7bb3f09e97732207625